dango/client/src/signer.rs

```diff
@@ -168,12 +216,51 @@ impl Signer for SingleSigner<Defined<u32>> {
             messages: msgs.clone(),
             data: metadata.clone(),
         };
-        let sign_data = sign_doc.to_sign_data()?;
 
-        let credential = Credential::Standard(StandardCredential {
-            key_hash: self.key_hash,
-            signature: Signature::Secp256k1(self.sk.sign_digest(sign_data.into()).into()),
-        });
+        let credential = match self.key {
+            Key::Secp256k1(_) => {
+                let sign_data = sign_doc.to_sign_data()?;
+
+                Credential::Standard(StandardCredential {
+                    key_hash: self.key_hash,
+                    signature: Signature::Secp256k1(self.sk.sign_digest(sign_data.into()).into()),
+                })
+            },
+            Key::Ethereum(_) => {
+                let verifying_contract =
+                    Some(U160::from_be_bytes(sign_doc.sender.into_inner()).into());
```

Comment:
We can `impl From<Addr> for alloy::primitive::U160` instead. Put this impl in `grug_types::eth_utils` as well.